blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -135,11 +143,33 @@ private boolean isReplayable(ReplayAction replayAction) {
         return replayAction != null && replayAction.isRebuildEnabled();
     }
 
+    static final long PIPELINE_NODE_CONTAINER_CACHE_MAX_SIZE = Long.getLong("PIPELINE_NODE_CONTAINER_CACHE_MAX_SIZE", 10000);
+
+    private static Cache<String, PipelineNodeContainerImpl> PIPELINE_NODE_CONTAINER_LOADING_CACHE = Caffeine.newBuilder()
```

Comment:
How large is the retained size of a `PipelineNodeContainerImpl`? Does it refer to any objects from other plugins esp. `workflow-*`?

Reply:
the reference to WorkflowRun has been removed here ae3da2701919e5d6c4dfa1fecd054754f247e2d8 (was not really useful) but some objects a bit deeper in the tree keep reference on it...

Reply:
Probably wise to avoid all `WorkflowRun` references from cache values, since they can be big and are supposed to be possible to unload.

Reply:
Yes I definitely agree but it's lot of changes for a bug branch.. anyway we have to do it but I'm not exactly sure how to load? Jenkins.get().getItem? But what is the key/id here?

Reply:
For a `Job`, use `.fullName`; for a `Run`, use `.externalizableId`.

Reply:
uhm trying to load from `getExternalizableId ` doesn't work.
A quick test such `Jenkins.get().getItem(run.getExternalizableId())` returns null.

Reply:
I have some other ways to do it but my concern is to not go to some list search etc... which could potentially load some data we don't need.

Reply:
> returns null

Because you are mixing up `Job` and `Run` ids. The first two methods for jobs. The second two for builds.